### ISSUE TITLE
LegendItem.removeItem() does nothing

### ISSUE THREAD

**Issue by kangalioo** [pyqtgraph/pyqtgraph#1041]
### Short description
The `LegendItem.removeItem(item)` function does nothing when called.

### Code to reproduce
```python
import pyqtgraph as pg

# Open plot window
win = pg.plot()

# Add legend
legend = win.addLegend()

# Generate two plots with names 'plot1' and 'plot2'
plot1 = win.plot([1,2,3], [2,3,5], name="plot1")
plot2 = win.plot([1,2,3], [3,5,6], name="plot2")

# Remove plot1. A sane default of pyqtgraph would be to automatically
# delete its legend entry as well (that's just a nice to have though)
win.removeItem(plot1)

# However, it's not even possible to explicitly delete the legend entry
# from the legend. This does nothing at all
legend.removeItem(plot1)

# Now there is one line, but two legend entries.
pg.QtGui.QApplication.exec_()
```

### Expected behavior
The legend entry should disappear when the `removeItem` function is called with the correct item.

Even nicer would be if the legend was automatically synced to the plot, i.e. calling `PlotItem.removeItem()` also removes the corresponding from the legend.

### Real behavior
Calling `LegendItem.removeItem(item)` does nothing at all as far as I can see.

### Tested environment(s)

 * PyQtGraph version: 0.10.0
 * Qt Python binding: PyQt5 5.13.1 Qt 5.13.1
 * Python version: 3.7.4
 * NumPy version: 1.16.4
 * Operating system: Antergos Linux 5.2.14
 * Installation method: pip3

### Additional context
Is there a nice workaround? I've fiddled around a bit (deleting and re-adding the legend entry to rebuild it from scratch for each entry deleted), but couldn't get anything to work.

**Comment by ixjlyons** [comment]
This was sort of a silent failure because at the time of 0.10.0, `removeItem` expected the string name of the plot and no error is raised if there's no match. #584 made it possible to pass the item itself instead, so I think we can close this.

By the way, since you seem to have some interest in pyqtgraph and reporting issues, it would be helpful if you could [install from the develop branch](https://github.com/pyqtgraph/pyqtgraph#installation-methods) because a ton of stuff has been fixed since 0.10.0 came out several years ago.

**Comment by kangalioo** [comment]
Oh, that's fantastic, I didn't know there was a more recent version of PyQtGraph! Gonna install that right tomorrow
